src/NuGet.Core/NuGet.Protocol/Plugins/PluginManager.cs

```diff
@@ -0,0 +1,368 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using NuGet.Common;
+using NuGet.Packaging;
+using NuGet.Protocol.Plugins;
+using NuGet.Shared;
+
+namespace NuGet.Protocol.Core.Types
+{
+    /// <summary>
+    /// A plugin manager. This manages all the live plugins and their operation claims.
+    /// Invoked in by both the credential provider and the PluginResourceProvider
+    /// </summary>
+    /// <remarks>This is unsealed only to facilitate testing.</remarks>
```

Comment:
Is this still true?  If no, remove and mark class `sealed`.

Reply:
marked it as sealed.